libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -211,10 +211,11 @@ pub struct NotificationHandler {
 }
 
 impl NotificationHandler {
-    pub fn listen(&self) -> Result<CmNotification, ConnectionManagerError> {
-        match self.recv.recv() {
-            Ok(notifications) => Ok(notifications),
-            Err(_) => Err(ConnectionManagerError::SendMessageError(
+    pub fn try_next(&self) -> Result<Option<CmNotification>, ConnectionManagerError> {
```

Comment:
What's the justification for this?  There's not information in the commit message explaining why this is the case.  

It seems like it may just introduce busy-wait situation for the user.  Shouldn't there still be the blocking version of this method?